packages/protocol/contracts/governance/ReleaseGoldInstance.sol

```diff
@@ -0,0 +1,579 @@
+   * @param _canValidate If this schedule's gold can be used for validating.
+   * @param _canVote If this schedule's gold can be used for voting.
+   * @param registryAddress Address of the deployed contracts registry.
+   */
+  constructor(
+    uint256 releaseCliffTime,
+    uint256 numReleasePeriods,
+    uint256 releaseStartTime,
+    uint256 releasePeriod,
+    uint256 amountReleasedPerPeriod,
+    bool revocable,
+    address payable _beneficiary,
+    address payable _releaseOwner,
+    bool subjectToLiquidityProvision,
+    bool _canValidate,
+    bool _canVote,
+    address registryAddress
+  ) public {
+    require(numReleasePeriods >= 1, "There must be at least one releasing period");
+    require(
+      amountReleasedPerPeriod > 0,
+      "The released amount per period must be greater than zero"
+    );
+    require(
+      _beneficiary != address(0),
+      "The release schedule beneficiary cannot be the zero addresss"
+    );
+    require(
+      _releaseOwner != address(0),
+      "The release schedule releaseOwner cannot be the zero address"
+    );
+    require(registryAddress != address(0), "The registry address cannot be the zero address");
+    require(
+      releaseStartTime.add(numReleasePeriods.mul(releasePeriod)) > block.timestamp,
+      "Release schedule end time must be in the future"
+    );
+
+    setRegistry(registryAddress);
+
+    releaseSchedule.numReleasePeriods = numReleasePeriods;
+    releaseSchedule.amountReleasedPerPeriod = amountReleasedPerPeriod;
+    releaseSchedule.releasePeriod = releasePeriod;
+    releaseSchedule.releaseCliff = releaseStartTime.add(releaseCliffTime);
+    releaseSchedule.releaseStartTime = releaseStartTime;
+
+    beneficiary = _beneficiary;
+    revocationInfo.revocable = revocable;
+    releaseOwner = _releaseOwner;
+
+    // Set maxDistribution default to maxUInt, i.e. no set max.
+    maxDistribution = ~uint256(0);
+    liquidityProvisionMet = (subjectToLiquidityProvision) ? false : true;
+    canValidate = _canValidate;
+    canVote = _canVote;
+  }
+
+  /**
+   * @notice Returns if the release schedule has been revoked or not.
+   * @return True if instance revoked.
+   */
+  function isRevoked() public view returns (bool) {
+    return revocationInfo.revokeTime > 0;
+  }
+
+  /**
+   * @notice Controls if the liquidity provision has been met.
+   *         If true, allows gold to be withdrawn.
+   */
+  function setLiquidityProvision(bool met) external onlyReleaseOwner {
+    liquidityProvisionMet = met;
+    emit LiquidityProvisionSet(beneficiary, met);
+  }
+
+  uint256 private constant FIXED1_UINT = 1000000000000000000000000;
+
+  /**
+   * @notice Controls the maximum distribution percentage.
+   *         Calculates `distributionPercentage` of current `totalBalance()`
+   *         and sets this value as the maximum allowed gold to be currently withdrawn.
+   */
+  function setMaxDistribution(uint256 distributionPercentage) external onlyReleaseOwner {
```

Comment:
Should we require distributionPercentage to be less than or equal to 100%?

Reply:
We can, I'm currently relying on Fixidy's overflow protection to prevent any issues from >100%. Other than that, maxDistribution can only limit how much you can withdraw, it can't add to how much you withdraw.
I'll add <= 100% to be clear